src/schemata/definitions/field.yml

```diff
@@ -2,6 +2,106 @@
 title: Field
 description: |
   A valid field schema is either a valid JSON schema as per [Draft 7](http://json-schema.org/draft-07/schema#) or one of our custom field types. Some additional properties are allowed at the top level.
+not:
+  oneOf:
+    # only allow `ui:options.rows` for string fields
+    - required: ['type', 'ui:options']
```

Comment:
why is it applied as a negative logic?? how would it looks like on the rendered documentation schema?

Reply:
How else can we make this work? This, for example, is what we do want to support:

```
type:
  const: boolean
ui:widget:
  type: string
  enum: ['checkbox','switch']
 ```
If we nest this in `oneOf` we also need to add a case for all the possible valid combinations for a boolean field, which cannot include in this combination of settings though, must have `ui:widget` as optional, and must support the other custom properties and the Draft 07 ones. 🤔 

If we nest this in `allOf`... well then I don't see how we can exclude anything

Reply:
I refactored it, we are now left with just one `not` case. This will require changes on the schema docs, since we have not used the `if` and `then` keywords before.